shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs

```diff
@@ -410,27 +410,32 @@ putWord7s (Word7 x : xs) = B.putWord8 (x .|. 0x80) >> putWord7s xs
 getWord7s :: Get [Word7]
 getWord7s = do
   next <- B.getWord8
-  case next .&. 0x80 of
-    0x80 -> (:) (toWord7 next) <$> getWord7s
-    _ -> pure [Word7 next]
-
-natToWord7s :: Natural -> [Word7]
-natToWord7s = reverse . go
+  -- is the high bit set?
+  if testBit next 7
+  -- if so, grab more words
+  then (:) (toWord7 next) <$> getWord7s
+  -- otherwise, this is the last one
+  else pure [Word7 next]
```

Comment:
I have tried limiting this and its CompactAddr sibling to only reading 8 `Word7`s (assuming each Word7 takes up 1 byte and not 7 bits), but for some reason that makes the `propCompactAddrRoundTrip` test fail. hopefully, this doesn't actually matter?

Reply:
The bound this changeset introduces is not respected by the generator for `Ptr`.

See `Test.Shelley.Spec.Ledger.Serialisation.EraIndepGenerators`

Reply:
since `Ptr` is now backed by `Word64` and not `Natural`, surely the generic instances won't generate anything larger than a Word64.

```hs
instance Arbitrary Ptr where
  arbitrary = genericArbitraryU
  shrink = genericShrink
```

this looks alright to me

Reply:
Yeah you're right.

Maybe this decoder can't handle the full word64 range?

(Also, why is Word64 the chosen limit here?)

Reply:
after some conversation with @redxaxder the `Ix` type will remain backed by `Natural` but decoding will be limited to 64 bytes, I was a bit confused by the jira issue that spawned this

Reply:
yea, my apologies about that jira ticket.